is.js

```diff
@@ -85,6 +85,75 @@
         };
     }
 
+    // helper function retrieves object keys
+    var objectKeys = function(object) {
+        if(Object.keys) {
+            return Object.keys(object);
+        }
+        var keys = [],
+            key;
+        for(key in object) {
+            if(hasOwnProperty.call(object, key)) {
+                keys.push(key);
+            }
+        }
+        return keys;
+    };
```

Comment:
Could you factor out the `objectKeys` helper and avoid the keys length early exit.